### PR TITLE
Use separate method to get next lexer token in parser

### PR DESCRIPTION
**Pull request by nijel**
This allows to customize the behavior in subclasses while keeping the
main parser logic.

We're customizing the parser to track lexer tokens in translate-toolkit and right now we have to copy whole `token` method. With this patch all we need is to customize `next_lexer_token`. See https://github.com/translate/translate/pull/3697

### CODE REVIEW

File: phply/phplex.py [hunk 1]
```diff
@@ -515,8 +515,11 @@ def current_state(self):
     def input(self, input):
         self.lexer.input(input)
 
+    def next_lexer_token(self):
```
Comment (viraptor) [inline phply/phplex.py]: Could you add a comment about what this is useful for? (both so others know and so it doesn't get removed as not useful in the future)

Comment (nijel) [reply phply/phplex.py]: Good point, docs added.

**Comment by nijel** [comment]
Thanks for merging! Any plans for release in near future?

**Comment by viraptor** [comment]
Yup. Going to deal with https://github.com/viraptor/phply/issues/43 in the next few days and make a release then.

**Comment by nijel** [comment]
Great, I'll adjust translate-toolkit then to use this.

**Pull request merged**